### PR TITLE
[DO NOT MERGE] error codes 21.02

### PR DESCRIPTION
**Pull request by jbkyang-nvi**

### CODE REVIEW

File: src/servers/http_server.cc [hunk 1]
```diff
@@ -1111,6 +1112,22 @@ class HTTPAPIServer : public HTTPServerImpl {
   re2::RE2 cudasharedmemory_regex_;
 };
 
+void HTTPAPIServer::HandleInferErrors(evhtp_request_t* req, TRITONSERVER_Error* err) {
+  const char* message = TRITONSERVER_ErrorMessage(err);
+  std::string error_string(message);
+  EVBufferAddErrorJson(req->buffer_out, err);
+  if (error_string.find("GPU error occured") == std::string::npos) {
```
Comment (GuanLuo) [inline src/servers/http_server.cc:1119]: `!=`?

Comment (GuanLuo) [reply src/servers/http_server.cc:1119]: Should check with the customer on the exact error message to check

File: src/servers/http_server.cc [hunk 2]
```diff
@@ -2325,13 +2342,10 @@ HTTPAPIServer::HandleInfer(
 
   if (err != nullptr) {
     LOG_VERBOSE(1) << "Infer failed: " << TRITONSERVER_ErrorMessage(err);
-    EVBufferAddErrorJson(req->buffer_out, err);
-    evhtp_send_reply(req, EVHTP_RES_BADREQ);
+    HandleInferErrors(req, err);
```
Comment (GuanLuo) [inline src/servers/http_server.cc:2345]: Control reaches here only if there is error before the model executes the request, so also need the check after `TRITONSERVER_InferenceResponseError()` where the error during execution will be reported to

File: src/servers/http_server.cc [hunk 3]
```diff
@@ -2325,13 +2342,10 @@ HTTPAPIServer::HandleInfer(
 
   if (err != nullptr) {
     LOG_VERBOSE(1) << "Infer failed: " << TRITONSERVER_ErrorMessage(err);
```
Comment (rmccorm4) [inline src/servers/http_server.cc:2344]: You can move this into the helper too, and doesn't need to call TRITONSERVER_ErrorMessage again, you can re-use `message`.

**Review by rmccorm4 — COMMENTED** [review]
Agree with Guan's comments as well.

File: src/servers/http_server.cc (hunk 1 above)
Comment (jbkyang-nvi) [reply src/servers/http_server.cc]: I confirmed with customer

File: Dockerfile.sdk [hunk 4]
```diff
@@ -126,8 +129,7 @@ RUN pip3 install --upgrade numpy pillow && \
     xargs pip3 install --upgrade
 
 # Install DCGM
-RUN wget https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/x86_64/datacenter-gpu-manager_${DCGM_VERSION}_amd64.deb && \
-    dpkg -i datacenter-gpu-manager_${DCGM_VERSION}_amd64.deb
+RUN apt-get update && apt-get install -y datacenter-gpu-manager=1:${DCGM_VERSION}
```
Comment (jbkyang-nvi) [inline Dockerfile.sdk:132]: can ignore these changes. These are for building the container with the recent cuda changes.

File: src/servers/http_server.cc [hunk 5]
```diff
@@ -2452,9 +2497,16 @@ HTTPAPIServer::InferRequestClass::InferResponseComplete(
...
-    TRITONSERVER_ErrorDelete(err);
-    evthr_defer(infer_request->thread_, BADReplyCallback, infer_request);
+    std::string msg(TRITONSERVER_ErrorMessage(err));
+    if (InferErrorIsInternalError(msg)) {
+      EVBufferAddErrorJson(infer_request->req_->buffer_out, err);
+      TRITONSERVER_ErrorDelete(err);
+      evthr_defer(infer_request->thread_, FAILReplyCallback, infer_request);
+    } else {
+      EVBufferAddErrorJson(infer_request->req_->buffer_out, err);
+      TRITONSERVER_ErrorDelete(err);
```
Comment (GuanLuo) [inline src/servers/http_server.cc:2507]: Seems like you can move this two lines out of the conditional:
```
EVBufferAddErrorJson...
TRITONSERVER_ErrorDelete...
if (InferErrorIsInternalError(msg)) {
   FAILReplyCallback
} else {
   BADReplyCallback
}
```

**Comment by rmccorm4** [comment]
LGTM. 

Not approving to avoid accidental merge.

File: Dockerfile.sdk (hunk 4 above)
Comment (jbkyang-nvi) [reply Dockerfile.sdk:132]: @pragyaarora

**Comment by rmccorm4** [comment]
Closing, not needed anymore after reviews.

**Pull request closed without merging**